lib/jose-jwe-decrypt.js

```diff
@@ -19,11 +19,15 @@
  *
  * @param cryptographer  an instance of WebCryptographer (or equivalent). Keep
  *                       in mind that decryption mutates the cryptographer.
- * @param key_promise    Promise<CryptoKey>, either RSA or shared key
+ * @param key    CryptoKey, Promise<CryptoKey>, either RSA or shared key
  */
-JoseJWE.Decrypter = function(cryptographer, key_promise) {
+JoseJWE.Decrypter = function(cryptographer, key) {
   this.cryptographer = cryptographer;
-  this.key_promise = key_promise;
+  // Identify JWK by kty property, try to import it
+  // or convert any passed key value into a promise
+  this.key_promise = (key.kty) ? Jose.Utils.importRsaPrivateKey(key) : new Promise(function (resolve) {
```

Comment:
I'm not sure about the instance you mean, but I've added a promise check first. And excluded a JWK import, because it seems that I'm far from the problems with JWK import at the moment. 

For example, I can't understand why in some cases JWK is imported like:
```
      cryptographer.setKeyEncryptionAlgorithm("A256KW");
      var shared_key = {"kty":"oct", "k":"GawgguFyGrWKav7AX4VKUg"};
      shared_key = crypto.subtle.importKey("jwk", shared_key, cryptographer.key_encryption.id, true, ["wrapKey", "unwrapKey"]);
```
while in others like:
```
var private_rsa_key = Jose.Utils.importRsaPrivateKey(rsa_key, "RSA-OAEP");
```